File: pacific-architecture/guava/src/main/java/com/pacific/guava/domain/SourceException.java
package com.pacific.guava.domain;

public class SourceException extends RuntimeException {

    public final int code;

    public SourceException(int code) {
        super();
        this.code = code;
    }

    public SourceException(String s, int code) {
        super(s);
        this.code = code;
    }

    public SourceException(String s, Throwable throwable, int code) {
        super(s, throwable);
        this.code = code;
    }

    public SourceException(Throwable throwable, int code) {
        super(throwable);
        this.code = code;
    }
}

File: pacific-architecture/guava/src/main/java/com/pacific/guava/domain/SourceException.kt
package com.pacific.guava.domain

class SourceException : RuntimeException {

    @get:JvmName("code")
    val code: Int

    constructor(code: Int) : super() {
        this.code = code
    }

    constructor(message: String?, code: Int) : super(message) {
        this.code = code
    }

    constructor(throwable: Throwable?, code: Int) : super(throwable) {
        this.code = code
    }

    constructor(message: String?, throwable: Throwable? , code: Int) : super(message, throwable) {
        this.code = code
    }
}